client/landing/stepper/hooks/use-stripe-connect-url.ts

```diff
@@ -0,0 +1,54 @@
+import { Onboard } from '@automattic/data-stores';
+import { isNewsletterFlow } from '@automattic/onboarding';
+import { useSelect } from '@wordpress/data';
+import { useEffect, useState } from 'react';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import wpcom from 'calypso/lib/wp';
+import type { OnboardSelect } from '@automattic/data-stores';
+
+type StripeConnectUrl = string | null;
+
+type MembershipsData = {
+	connect_url: string | undefined;
+	connected_account_default_currency: string | undefined;
+	connected_account_description: string | undefined;
+	connected_account_id: string | undefined;
+};
+
+const fetchMembershipsData = (
+	siteId: number,
+	source: string | null | undefined
+): Promise< MembershipsData > => {
+	const url = source
+		? `/sites/${ siteId }/memberships/status?source=${ source }`
+		: `/sites/${ siteId }/memberships/status`;
+	return wpcom.req.get( url, { apiNamespace: 'wpcom/v2' } );
+};
+
+export function useStripeConnectUrl(
+	siteId: number | undefined,
+	flow: string | null,
+	source?: string | null | undefined
+): StripeConnectUrl {
+	const [ stripeConnectUrl, setStripeConnectUrl ] = useState< StripeConnectUrl >( null );
+
+	const goals = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getGoals(),
+		[]
+	);
+
+	const hasSubscriberGoal = goals.includes( Onboard.SiteGoal.PaidSubscribers );
+
+	useEffect( () => {
+		// Only make this extra request if we know we need
+		// the stripe connect url - for newsletters sites
+		// with paid-subscribers goal
```

Comment:
I don't understand the logic, the hook is to fetch "stripe connect URL" but here we note that we need it only for Newsletter flow?

The `isNewsletterFlow` feels overly specific for a hook. Should we instruct the hook to do this all without passing the flow name?

Reply:
I've refactored this away from using a custom hook, and just defined a method for the API request, which is called in a useEffect if the site flow is newsletter.